pandora-server-shard/src/networking/manager_client.ts

```diff
@@ -414,6 +416,236 @@ export const ConnectionManagerClient = new class ConnectionManagerClient impleme
+										});
+										break; //Done on purpose
+									}
+									const card = space.cardGame.dealTo(game.action.targetId);
+									if (!card) {
+										space.handleActionMessage({
+											id: 'gamblingCardGameEmpty',
+											sendTo: [client.character.id],
+										});
+									} else {
+										//Dealt to the player openly
+										space.handleActionMessage({
+											id: 'gamblingCardGameDealPlayerOpen',
+											character,
+											sendTo: space.cardGame.getPlayerIds(), // Send messages only to players
+											dictionary: {
+												'CARD': card.toString(),
+											},
+										});
+									}
+								} else {
+									space.handleActionMessage({
+										id: 'gamblingCardGameNotAllowed',
+										sendTo: [client.character.id],
+										character,
+									});
+								}
+								break;
+							}
+							case 'deal': {
+								if (space.cardGame.isDealer(client.character.id)) {
+									// Deals a hidden card to a character
+									if (!space.cardGame.checkPlayer(game.action.targetId)) {
+										space.handleActionMessage({
+											id: 'gamblingCardNotAPlayer',
+											sendTo: [client.character.id],
+										});
+										break; //Done on purpose
+									}
+									const card = space.cardGame.dealTo(game.action.targetId);
+									if (!card) {
+										space.handleActionMessage({
+											id: 'gamblingCardGameEmpty',
+											sendTo: [client.character.id],
+										});
+									} else {
+										space.handleActionMessage({
+											id: 'gamblingCardGameDealPlayerSecret',
+											character,
+											// target: space.getCharacterById(game.action.targetId),
+											sendTo: space.cardGame.getPlayerIds(), // Send messages only to players
+											dictionary: {
+												'TARGET_CHARACTER': `${space.getCharacterById(game.action.targetId)?.name}`,
+											},
+										});
+										space.handleActionMessage({
+											id: 'gamblingCardGameDealToYou',
+											character,
+											sendTo: [game.action.targetId],
+											dictionary: {
+												'CARD': card.toString(),
+											},
+										});
+									}
+								} else {
+									space.handleActionMessage({
+										id: 'gamblingCardGameNotAllowed',
+										sendTo: [client.character.id],
+										character,
+									});
+								}
+								break;
+							}
+							case 'check': {
+								const spaceHand = space.cardGame.getSpaceHand();
+								space.handleActionMessage({
+									id: 'gamblingCardGameHandCheck',
+									character,
+									sendTo: [client.character.id],
+									dictionary: {
+										'HAND': space.cardGame.getPlayerHand(client.character.id),
+										'ISARE': (spaceHand === 'nothing' || spaceHand.length < 2) ? 'is' : 'are',
+										'TABLE': spaceHand,
+									},
+								});
+								break;
+							}
+							case 'reveal': {
+								if (space.cardGame.isDealer(client.character.id)) {
```

Comment:
Admin should be able to always stop the game, to handle cases like dealer going AFK.